### PR TITLE
Change var to let/const

### PR DESCRIPTION
**Pull request by arv**
Strong mode does not allow var declarations.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/syntax/unicode-tables.js [hunk 1]
```diff
@@ -39,13 +39,14 @@ export var idStartTable = [
   886, 887,
   890, 890,
   891, 893,
+  895, 895,
```
Comment (arv) [inline src/syntax/unicode-tables.js]: I guess the data at http://www.unicode.org/Public/UNIDATA/DerivedCoreProperties.txt got updated.

File: src/syntax/Scanner.js [hunk 2]
```diff
@@ -90,17 +90,8 @@ import {
...
-  isWhitespaceArray[i] = i >= 9 && i <= 13 ||  // Tab - Carriage Return
-      i === 0x20;  // Space
-}
-
-// Some of these is* functions use an array as a lookup table for the lower 7
-// bit code points.
-
-var isWhitespaceArray = [];
-for (var i = 0; i < 128; i++) {
+let isWhitespaceArray = [];
```
Comment (arv) [inline src/syntax/Scanner.js]: Caught a bug!

File: src/syntax/Parser.js [hunk 3]
```diff
@@ -3479,16 +3479,13 @@ export class Parser {
...
 
   /**
-   * @param {Initializer=} initializer If left out the initializer is
-   *     optional and allowed. If set to Initializer.REQUIRED there must be an
-   *     initializer.
    * @return {ParseTree}
    */
-  parseBindingElement_(initializer = Initializer.OPTIONAL) {
-    var start = this.getTreeStartLocation_();
+  parseBindingElement_() {
```
Comment (arv) [inline src/syntax/Parser.js]: This was never called with an argument.

**Comment by arv** [comment]
I apologize for the size of this. I have verified that this works with 'use strong' in my strong mode branch.

**Comment by johnjbarton** [comment]
"Alas, poor var! I knew him, Horatio; a fellow of infinite jest, ..."

LGTM

**Comment by arv** [comment]
FWIW, runtime and node still use var. I believe most of runtime can be changed since we compile the runtime these days.

**Pull request merged**